product_profile/models/product.py

```diff
@@ -0,0 +1,328 @@
+    ):
+        """ Display a warning for end user if edit record """
+        res = super(ProductProfile, self).fields_view_get(
+            view_id=view_id,
+            view_type=view_type,
+            toolbar=toolbar,
+            submenu=submenu,
+        )
+        if view_type == "form":
+            style = "alert alert-warning oe_text_center oe_edit_only"
+            alert = etree.Element("h2", {"class": style})
+            alert.text = _(
+                "If you update this profile, all products "
+                "using this profile could also be updated. "
+                "Changes can take a while."
+            )
+            doc = etree.XML(res["arch"])
+            doc[0].addprevious(alert)
+            res["arch"] = etree.tostring(doc, pretty_print=True)
+        return res
+
+
+class ProductMixinProfile(models.AbstractModel):
+    _name = "product.mixin.profile"
+
+    @api.model
+    def _get_profile_fields(self):
+        fields_to_exclude = set(get_profile_fields_to_exclude())
+        return [
+            field
+            for field in self.env["product.profile"]._fields.keys()
+            if field not in fields_to_exclude
+        ]
+
+    @api.model
+    def _get_vals_from_profile(self, values):
+        profile_obj = self.env["product.profile"]
+        fields = self._get_profile_fields()
+        vals = profile_obj.browse(values["profile_id"]).read(fields)[0]
+        processed_vals = deepcopy(vals)
+        for field, value in vals.items():
+            unprefixed_field = ""
+            if PROF_DEFAULT_STR == field[:LEN_DEF_STR]:
+                unprefixed_field = field[LEN_DEF_STR:]
+                if unprefixed_field not in values:
+                    # we only put the default profile value
+                    # if their is no matching in default data
+                    processed_vals[unprefixed_field] = vals[field]
+                # prefixed fields must be removed from dict
+                # because they are in profile not in product
+                processed_vals.pop(field)
+            if (
+                value
+                and profile_obj._fields[field].type
+                == "many2one"
+            ):
+                # m2o value is a tuple
+                processed_vals[unprefixed_field or field] = value[0]
+            if (
+                profile_obj._fields[field].type
+                == "many2many"
+            ):
+                processed_vals[unprefixed_field or field] = [(6, 0, value)]
+        return processed_vals
+
+    @api.onchange("profile_id")
+    def _onchange_from_profile(self):
+        """ Update product fields with product.profile corresponding fields """
+        self.ensure_one()
+        if self.profile_id:
+            values = self._get_vals_from_profile(
+                {"profile_id": self.profile_id.id}
+            )
+            for field, value in values.items():
+                try:
```

Comment:
i think is necessary to avoid  id that generate error message on change profile on product

```suggestion
                if field != 'id':
                        try:
                            self[field] = value
```

Reply:
Hi EBII, thank you for your feedback!

I think that field should be excluded already because of [https://github.com/kevinkhao/product-attribute/blob/701c805efa27ed7ac7b2df4d1f242ccef217957f/product_profile/models/product.py#L39](url) , did you manage to trigger an error message ?

Reply:
Yes @kevinkhao it generate the error message in the except 

get_profile_fields_to_exclude() is only call on write method.. You can call it On the on change instead of just filter the id.
but not sure it is necessary or it's work
```suggestion
excludable_fields = get_profile_fields_to_exclude()
for field, value in values.items():
    if field not in excludable_fields:
                try:
```